ouroboros-consensus/demo-playground/Mock/TxSubmission.hs

```diff
@@ -102,27 +104,34 @@ submitTx n tx = do
 -- | Auxiliary to 'spawnMempoolListener'
 readIncomingTx :: RunDemo blk
                => Tracer IO String
+               -> Tracer IO MemPoolSize
                -> NodeKernel IO peer blk
                -> Decoder IO
                -> IO ()
-readIncomingTx tracer kernel Decoder{..} = forever $ do
+readIncomingTx tracer mempoolTracer kernel Decoder{..} = forever $ do
     newTx :: Mock.Tx <- decodeNext decode
-    rejected <- addTxs (getMempool kernel) [demoMockTx (getNodeConfig kernel) newTx]
+    let memPool = getMempool kernel
+    rejected <- addTxs memPool [demoMockTx (getNodeConfig kernel) newTx]
+    txs <- STM.atomically $ getTxs memPool
+    traceWith tracer $
+      "New incoming tx, txs number in a mempool: " <> show (Seq.length txs)
+    traceWith mempoolTracer $ AfterAddTxs $ Seq.length txs
```

Comment:
@365andreas after merging #659 I think this changes should go to `Ouroboros.Cosensus.TxSubmission`